### PR TITLE
Add support for alpha phase

### PR DESCRIPTION
**Pull request by laurentqro**
Screenshot 2020-05-01 at 15 15 31

Screenshot 2020-05-01 at 15 16 36

https://trello.com/c/ZITqImYk/109-change-publisher-app-to-allow-publisher-to-show-which-phase-the-product-is-in

### CODE REVIEW

File: app/views/shared/_common_edition_attributes.html.erb [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
-      as: :boolean,
-      label: 'Content is in beta',
-      input_html: { disabled: @resource.locked_for_edits? },
-      wrapper_html: { class: "emphasised-field form-group add-bottom-margin input-md-7" } %>
+<div class="form-group">
+  <%= f.label :phase %>
+  <%= f.select :phase, Edition::PHASE_NAMES, {},
+    :class => "form-control input-md-3",
+    :disabled => @resource.locked_for_edits? %>
+</div>
```
Comment (kevindew) [inline app/views/shared/_common_edition_attributes.html.erb:19]: I wonder if we may need to explain what a phase is to users? It feels like a not very explanatory term. I imagine it'd be good to have hint text as per meta description, maybe we can get our content designer to help?

File: app/views/root/_publication.html.erb [hunk 2]
```diff
@@ -5,8 +5,8 @@
   <td class="title">
     <h4 class="publication-table-title">
       <%= link_to publication.admin_list_title, edition_path(publication) %>
-      <% if publication.in_beta? %>
-        <span class="badge badge-beta">beta</span>
+      <% if !publication.live? %>
+        <span class="badge badge-beta"><%= publication.phase %></span>
```
Comment (benthorner) [inline app/views/root/_publication.html.erb:9]: We should customise the badge, depending on the phase.

File: app/presenters/formats/edition_format_presenter.rb [hunk 3]
```diff
@@ -14,17 +14,7 @@ def render_for_publishing_api(republish: false)
...
-      if edition.auth_bypass_id
-        fields[:access_limited] = { auth_bypass_ids: [edition.auth_bypass_id] }
-      end
-
-      if !edition.live? && edition.phase_message
-        fields[:phase_message] = edition.phase_message
-      end
-
-      fields
+      { access_limited: access_limited }.delete_if { |_key, value| value.blank? }
```
Comment (benthorner) [inline app/presenters/formats/edition_format_presenter.rb:17]: I'm finding this quite hard to understand. Could we use a different technique here?

```
{
   access_limited: access_limited,
   ...
}.compact
```

File: test/unit/presenters/formats/edition_format_presenter_test.rb [hunk 4]
```diff
@@ -74,7 +74,11 @@ def result
     should "include phase message if it's not live" do
       edition.expects(:live?).returns(false)
       edition.expects(:phase_message).twice.returns("phase message")
-      assert_equal "phase message", result[:phase_message]
+
+      phase_message = result[:details][:phase_message]
+
+      assert_equal "phase message", phase_message[:content]
```
Comment (benthorner) [inline test/unit/presenters/formats/edition_format_presenter_test.rb]: Minor: this test could go in the previous commit, since this commit is a refactor.

File: app/presenters/formats/edition_format_presenter.rb [hunk 5]
```diff
@@ -56,7 +46,24 @@ def document_type
...
     def details
-      {}
+      { phase_message: phase_message }.delete_if { |_key, value| value.blank? }
+    end
+
+    def phase_message
+      if show_phase_message?
+        {
+          content: edition.phase_message,
+          content_type: "text/html",
```
Comment (benthorner) [inline app/presenters/formats/edition_format_presenter.rb]: It looks like we're changing the format of the payload here, so it now includes a content type. Can you explain this in the commit message? (It could also be its own commit.)

File: app/views/shared/_common_edition_attributes.html.erb (hunk 1 above)
Comment (benthorner) [reply app/views/shared/_common_edition_attributes.html.erb:19]: Since we weren't giving any hint before, it seems OK continue like that. I do think "Development" might throw people. Perhaps "Content phase"?

File: test/unit/presenters/formats/edition_format_presenter_test.rb [hunk 6]
```diff
@@ -62,18 +64,10 @@ def result
...
-        assert_equal "beta", result[:phase]
-      end
-    end
-
-    context "when not in beta" do
-      should "not include phase" do
-        edition.expects(:in_beta).returns(false)
-        assert_not_includes result.keys, :phase
-      end
+    should "include phase name" do
```
Comment (benthorner) [inline test/unit/presenters/formats/edition_format_presenter_test.rb:67]: Are we missing a test here now?

Comment (laurentqro) [reply test/unit/presenters/formats/edition_format_presenter_test.rb:67]: you mean the "should not include phase" test?

Comment (benthorner) [reply test/unit/presenters/formats/edition_format_presenter_test.rb:67]: Yeah, what happened to that?

Comment (laurentqro) [reply test/unit/presenters/formats/edition_format_presenter_test.rb:67]: I'll add it back in.

Comment (laurentqro) [reply test/unit/presenters/formats/edition_format_presenter_test.rb]: done

File: app/controllers/editions_controller.rb [hunk 7]
```diff
@@ -319,6 +319,8 @@ def common_params
        title
        in_beta
```
[the remark this answers is not in the document]
Comment (laurentqro) [reply app/controllers/editions_controller.rb]: we'll keep this for the migration to run, and then we'll be able to remove it.

**Comment by vanitabarrett** [comment]
@laurentqro Can we close this PR?

**Comment by laurentqro** [comment]
yes, we can, I'll close it.
